.github/workflows/ci.yaml

```diff
@@ -0,0 +1,222 @@
+          args: --timeout=5m0s
+      - name: Run bootstrap
+        run: PIPELINE=1 script/bootstrap.sh
+      - run: make test
+      - name: 'Get Previous tag'
+        id: previoustag
+        uses: "WyriHaximus/github-action-get-previous-tag@v1"
+        with:
+          fallback: v0.0.0
+      - name: Set image tag
+        shell: bash
+        id: tags
+        run: |
+          if [[ ${{ github.ref_name }} =~ ^v.* ]] ; then
+            echo "VERSION=${{ github.ref_name }}" >> "${GITHUB_OUTPUT}"
+          else
+             echo "VERSION=${{ steps.previoustag.outputs.tag }}-dev" >> "${GITHUB_OUTPUT}"
+          fi
+      - name: Get image tag
+        run: |
+          echo ${{ steps.tags.outputs.VERSION }}
+      - name: iofogctl build packages GoReleaser
+        uses: goreleaser/goreleaser-action@v4
+        with:
+          version: 1.1.0
+          args: --snapshot --rm-dist --debug --config ./.goreleaser-iofogctl.yml
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: iofogctl
+          path: ${{ github.workspace }}/dist
+  Test:
+    needs: Build
+    runs-on: ubuntu-20.04
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+      packages: 'read'
+    steps:
+      - uses: actions/checkout@v3
+      - name: Login to Github Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: "ghcr.io"
+          username: ${{ github.actor }}
+          password: ${{ secrets.PAT }}
+      - name: 'Pull develop ghcr docker image'
+        run: |
+          docker pull ${{ env.controller_image }}
+          docker pull ${{ env.agent_image }}
+      - uses: actions/download-artifact@v2
+        with:
+         name: iofogctl
+      - run: ls
+      - name: Prepare iofogctl binary
+        shell: bash
+        run: |
+          sudo cp ${{ github.workspace }}//build_linux_linux_amd64/iofogctl /usr/local/bin/
+          sudo cp ${{ github.workspace }}//build_linux_linux_amd64/iofogctl /usr/local/bin/
+          sudo chmod 0755 /usr/local/bin/iofogctl
+      - run: sudo apt-get install -y jq
+      - name: Verify iofogctl
+        run: |
+          which iofogctl
+          iofogctl version
+      - run: ./pipeline/steps/configure-env.sh
+      - name: Install test deps
+        shell: bash
+        run: |
+          git clone https://github.com/bats-core/bats-core.git && cd bats-core && git checkout tags/v1.1.0 && sudo ./install.sh /usr/local
+          sudo npm i -g tap-junit
+      - name: Run smoke test
+        run: test/run.bash smoke
+      - name: Run functional Test
+        shell: bash
+        run: |
+          set -o pipefail
+          test/run.bash local | tee test/conf/results-local.tap
+      - name: 'Convert test output from TAP to JUnit'
+        shell: bash
+        run: |
+          tap-junit -i test/conf/results-local.tap -o test/conf -s Local -n results-local.xml || true
+      - name: 'Clean local docker'
```

Comment:
Not really required. You get a fresh docker cache on every run unless configured otherwise.

Reply:
Removed